### PR TITLE
Implement QuantizeLinear and DequantizeLinear

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by tianleiwu** [comment]
Add some tests

**Comment by yufenglee** [comment]
> Add some tests

CPU tests cover it.

File: onnxruntime/core/providers/cuda/tensor/quantize_linear.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              y_zero_point != nullptr);
+  auto y = ctx->Output(0, x->Shape());
+  ORT_ENFORCE(y != nullptr);
+
+  const auto& x_shape = x->Shape();
+
+  const float* input = x->template Data<float>();
+  T* output = y->template MutableData<T>();
+
+  ORT_ENFORCE(IsScalarOr1ElementVector(y_scale), "x_scale must be a scalar or 1D tensor or size 1.");
```
Comment (tianleiwu) [inline onnxruntime/core/providers/cuda/tensor/quantize_linear.cc]: of size 1

File: onnxruntime/core/providers/cuda/tensor/quantize_linear.cu [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+template <int NumThreadsPerBlock, int NumElementsPerThread>
+__global__ void QuantizeLinearKernel(const float* input, uint8_t* output, const float* scale, const uint8_t* zero_point, CUDA_LONG N) {
+  CUDA_LONG id = NumElementsPerThread * NumThreadsPerBlock * blockIdx.x + threadIdx.x;
+
+#pragma unroll
+  for (int i = 0; i < NumElementsPerThread; i++) {
+    if (id < N) {
+      int value = __float2int_rn(input[id] / (*scale)) + *zero_point;
```
Comment (tianleiwu) [inline onnxruntime/core/providers/cuda/tensor/quantize_linear.cu]: add static_cast<int> for zero_point since it is unsigned

**Pull request merged**